share/getters/shrex.go

```diff
@@ -79,6 +108,7 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 				"hash", root.String(),
 				"err", getErr,
 				"finished (s)", time.Since(start))
+			sg.metrics.attempts.Record(ctx, int64(attempt), attribute.Bool("success", false))
```

Comment:
If metrics are not enabled this will segfault. We usually wrap Record with a method on metrics start whic also checks whether metrics is nil

Reply:
this was just a prototype, forgot to extract it to its own file + do it correctly, thanks

Reply:
pls check again